### PR TITLE
JAMES-2928 Make GenericMailet's precondition policy consistent

### PR DESCRIPTION
**Pull request by blackheaven**
With a bonus fix on `WithPriority`

### CODE REVIEW

File: mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java [hunk 1]
```diff
@@ -82,9 +84,7 @@ public void destroy() {
      * @throws NullPointerException before {@link #init(MailetConfig)}
      */
     public boolean getInitParameter(String name, boolean defaultValue) {
-        if (config == null) {
-            throw new NullPointerException("Mailet configuration must be set before getInitParameter is called.");
-        }
+        Preconditions.checkNotNull(config, CONFIG_IS_NULL_ERROR_MESSAGE);
```
Comment (mbaechler) [inline mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java]: technically speaking, it should not be a null pointer exception but an illegal state exception

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithPriority.java [hunk 2]
```diff
@@ -50,17 +50,17 @@ public String getMailetInfo() {
     @Override
     public void init() throws MessagingException {
         Integer priorityRaw = Optional.ofNullable(getInitParameter("priority", null))
-                .map(Integer::valueOf)
-                .orElseThrow(() -> new IllegalArgumentException("'priority' init parameter is compulsory"));
+            .map(Integer::valueOf)
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithPriority.java]: why it's not detected by the build?

**Comment by remk** [comment]
```
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Running org.apache.james.mpt.DiscardProtocolTest
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 61.352 s <<< FAILURE! - in org.apache.james.mpt.DiscardProtocolTest
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [ERROR] testRecord  Time elapsed: 60.033 s  <<< ERROR!
[63ae855368ed0eb2d954c48c04d4ce12774403ae] java.lang.Exception: Timed out wait for be notified that read is complete
[63ae855368ed0eb2d954c48c04d4ce12774403ae] 	at org.apache.james.mpt.DiscardProtocolTest.testRecord(DiscardProtocolTest.java:84)
[63ae855368ed0eb2d954c48c04d4ce12774403ae] 
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Running org.apache.james.mpt.TestScriptedUserAdder
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.04 s - in org.apache.james.mpt.TestScriptedUserAdder
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Running org.apache.james.mpt.TestFileProtocolSessionBuilder
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.48 s - in org.apache.james.mpt.TestFileProtocolSessionBuilder
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Running org.apache.james.mpt.TestExternalHostSystem
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.035 s - in org.apache.james.mpt.TestExternalHostSystem
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Running org.apache.james.mpt.api.ImapFeaturesTest
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.015 s - in org.apache.james.mpt.api.ImapFeaturesTest
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] 
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] Results:
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [INFO] 
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [ERROR] Errors: 
[63ae855368ed0eb2d954c48c04d4ce12774403ae] [ERROR]   DiscardProtocolTest.testRecord:84 »  Timed out wait for be notified that read ...
```
test this please.

**Comment by Arsnael** [comment]
```
[ERROR]   WithEncryptedSwiftTest.mailsShouldBeWellReceived » ConditionTimeout Condition ...
```
test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**